### PR TITLE
feat(@clayui/drop-down): Render Menu only after trigger is clicked the first time

### PR DESCRIPTION
**Pull request by sandrodw3**
Related issue: https://github.com/liferay/clay/issues/4708

### CODE REVIEW

**Review by julien — COMMENTED** [review]
Left a comment requesting a small change, but the rest looks good to me.
@matuzalemsteles let me know what you think when you can - thanks.

File: packages/clay-drop-down/src/DropDown.tsx [hunk 1]
```diff
@@ -115,6 +115,8 @@ const ClayDropDown: React.FunctionComponent<IProps> & {
 	const triggerElementRef = React.useRef<HTMLButtonElement | null>(null);
 	const menuElementRef = React.useRef<HTMLDivElement>(null);
 
+	const [initialized, setInitialized] = useState(false);
```
[the remark this answers is not in the document]
Comment (julien) [reply packages/clay-drop-down/src/DropDown.tsx]: @sandrodw3 thanks :heart: !

**Comment by matuzalemsteles** [comment]
Oh, that's a good strategy, something like lazy. Well, I don't remember why we keep it that way but I remember we had some problems with it but maybe because we used the `active` prop but apparently I think it is related to the DropDown alignment once we render it in the body or the nearest sub portal.

But going with this solution doesn't seem to break it, it's a nice improvement @sandrodw3, thanks! We can also see other solutions if that doesn't help much, like centralizing a provider for DropDown to be a singleton like the Modal provider.

**Review by matuzalemsteles — APPROVED** [review]
LGTM!

**Comment by sandrodw3** [comment]
> Oh, that's a good strategy, something like lazy. Well, I don't remember why we keep it that way but I remember we had some problems with it but maybe because we used the `active` prop but apparently I think it is related to the DropDown alignment once we render it in the body or the nearest sub portal.
> 
> But going with this solution doesn't seem to break it, it's a nice improvement @sandrodw3, thanks! We can also see other solutions if that doesn't help much, like centralizing a provider for DropDown to be a singleton like the Modal provider.

Thanks for the feedback @matuzalemsteles 🥰. I have made some tests in portal with this solution and it completely solves our performance problem in Pages Admin, so I guess it's enough for now.

**Pull request merged**